components/data-feed-service/dao/destination.go

```diff
@@ -1,8 +1,175 @@
 package dao
 
+import (
+	"strconv"
+
+	datafeed "github.com/chef/automate/api/external/data_feed"
+	"github.com/chef/automate/lib/errorutils"
+	"github.com/pkg/errors"
+)
+
 type Destination struct {
 	ID     int64  `db:"id"`
 	Name   string `db:"name"`
 	URL    string `db:"url"`
 	Secret string `db:"secret"`
 }
+
+func (db *DB) addToDBDestination(inDestination *datafeed.AddDestinationRequest) *Destination {
+	newDestination := Destination{}
+	newDestination.ID = inDestination.Id
+	newDestination.Name = inDestination.Name
+	newDestination.URL = inDestination.Url
+	newDestination.Secret = inDestination.Secret
+
+	return &newDestination
+}
+
+func (db *DB) updateToDBDestination(inDestination *datafeed.UpdateDestinationRequest) *Destination {
+	newDestination := Destination{}
+	newDestination.ID = inDestination.Id
+	newDestination.Name = inDestination.Name
+	newDestination.URL = inDestination.Url
+	newDestination.Secret = inDestination.Secret
+
+	return &newDestination
+}
+
+func (db *DB) dbToGetDestinationResponse(inDestination *Destination) *datafeed.GetDestinationResponse {
+	newDestination := datafeed.GetDestinationResponse{}
+	newDestination.Id = inDestination.ID
+	newDestination.Name = inDestination.Name
+	newDestination.Url = inDestination.URL
+	newDestination.Secret = inDestination.Secret
+
+	return &newDestination
+}
+
+func (db *DB) getDestinationRequestToDB(inDestination *datafeed.GetDestinationRequest) *Destination {
+	newDestination := Destination{}
+	newDestination.ID = inDestination.Id
+
+	return &newDestination
+}
+
+func (db *DB) AddDestination(destination *datafeed.AddDestinationRequest) (bool, error) {
+	dbDestination := db.addToDBDestination(destination)
+	var err error
```

Comment:
[nit] I think we might want to avoid any confusion about shadowing here if possible by dropping this `var err error` and doing something like:

```
	err := Transact(db, func(tx *DBTrans) error {
		if err := tx.Insert(dbDestination); err != nil {
			return errors.Wrap(err, "AddDestination: unable to insert destination")
		}
		return nil
	})
```

similarly in the other examples like this.